test/extended/openstack/utils.go

```diff
@@ -0,0 +1,119 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+
+	o "github.com/onsi/gomega"
+)
+
+func ElementExists(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func RandomSuffix() string {
+	return strconv.Itoa(rand.Intn(10000))
+}
+
+func SubnetExists(networkClient *gophercloud.ServiceClient, listOpts subnets.ListOpts, ns *v1.Namespace) bool {
+	subnet_name := ns.Name
+	subnet_alt_name := "ns/" + ns.Name + "-subnet"
+	subnets := ListSubnets(networkClient, listOpts)
+	subnets_names := []string{}
+	for _, s := range subnets {
+		subnets_names = append(subnets_names, s.Name)
+	}
+	return ElementExists(subnets_names, subnet_name) || ElementExists(subnets_names, subnet_alt_name)
+}
+
+func ListSubnets(networkClient *gophercloud.ServiceClient, listOpts subnets.ListOpts) []subnets.Subnet {
+	allPages, err := subnets.List(networkClient, listOpts).AllPages()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	rSubnets, err := subnets.ExtractSubnets(allPages)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	return rSubnets
+}
+
+func CreateNamespace(clientSet *kubernetes.Clientset, baseName string) *v1.Namespace {
+	nsName := fmt.Sprintf("%v-%v", baseName, RandomSuffix())
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nsName,
+		},
+	}
+	clientSet.CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{})
+	e2e.Logf("Namespace %v was created", nsName)
+
+	return ns
+}
+
+func CreatePod(clientSet *kubernetes.Clientset, nsName string, baseName string, hostNetwork bool) {
+	podName := fmt.Sprintf("%v-%v", baseName, RandomSuffix())
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "demo",
+					Image: "quay.io/kuryr/demo",
+				},
+			},
+			HostNetwork: hostNetwork,
+		},
+	}
+	p, err := clientSet.CoreV1().Pods(nsName).Create(context.TODO(), pod, metav1.CreateOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(WaitForPodReady(clientSet, nsName, p.Name, 120*time.Second)).To(o.BeTrue())
```

Comment:
I think k8s e2e tests wait for 5 minutes, maybe we should stick to that.

Reply:
5 minutes seems a lot to me. If we'll see problems we can check and adjust. wdyt?

Reply:
Okay, but then let's import and use this constant: https://github.com/kubernetes/kubernetes/blob/b4aa9a189ead272a3f6dd62562f7a355932c57fa/test/e2e/framework/util.go#L95-L98

Reply:
BTW - I just found a function [1] doing the same in the regular e2e framework, would be best to reuse it instead of reinventing the wheel.

[1] https://github.com/kubernetes/kubernetes/blob/da8ddb7485004ec5a5dea9fcafd1d3937abf15f2/test/e2e/framework/pod/wait.go#L363-L367